File: src/contests/cg/puzzle/dart/euclids-algorithm.dart
// 🎮 CodinGame Puzzle - euclids-algorithm
// https://www.codingame.com/training/easy/euclids-algorithm

import 'dart:io';

void main() {
  var parts = stdin.readLineSync()!.trim().split(' ');
  int a = int.parse(parts[0]);
  int b = int.parse(parts[1]);
  int x = a, y = b;
  while (y != 0) {
    int q = x ~/ y;
    int r = x % y;
    print('$x=$y*$q+$r');
    x = y;
    y = r;
  }
  print('GCD($a,$b)=$x');
}

File: src/contests/cg/puzzle/perl/euclids-algorithm.pl
# 🎮 CodinGame Puzzle - euclids-algorithm
# https://www.codingame.com/training/easy/euclids-algorithm

my ($a, $b) = split ' ', <STDIN>;
my ($x, $y) = ($a, $b);
while ($y != 0) {
    my $q = int($x / $y);
    my $r = $x % $y;
    print "$x=$y*$q+$r\n";
    ($x, $y) = ($y, $r);
}
print "GCD($a,$b)=$x\n";
